api/src/main/java/com/cloud/deploy/DeployDestination.java

```diff
@@ -78,6 +78,10 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
+        if (obj == null)
+            return false;
+        if (!(obj instanceof DeployDestination))
+            return false;
         DeployDestination that = (DeployDestination)obj;
         if (_dc == null || that._dc == null) {
```

Comment:
Besides what @GabrielBrascher suggested, I have one suggestion though.
Can you append the word `this.` to the `_dc` variable? Then, we would have `this._dc == null || that._dc == null`.

Reply:
added `this.` as suggested